src/services/codefixes/convertTypedefToType.ts

```diff
@@ -0,0 +1,232 @@
+                typeReference = factory.createTypeLiteralNode(signatures);
+            }
+            // Leaf node, where type.kind === SyntaxKind.JSDocTypeExpression
+            else if (type) {
+                typeReference = createTypeReference(type);
+            }
+            if (typeReference && name) {
+                const prop = factory.createPropertySignature(
+                    [],
+                    name,
+                    // eslint-disable-next-line local/boolean-trivia
+                    undefined,
+                    typeReference
+                );
+                return [...signatures, prop];
+            }
+
+    }, []);
+    return props;
+
+}
+
+function getPropertyName(tag: JSDocPropertyTag): string | undefined {
+    const { name } = tag;
+    if (!name) return;
+
+    let propertyName;
+    // for "@property {string} parent.child" or "@property {string} parent.child.grandchild" in nested object type
+    // We'll get "child" in the first example or "grandchild" in the second example as the prop name
+    if (name.kind === SyntaxKind.QualifiedName) {
+        propertyName = name.right.getFullText().trim();
+    }
+    else {
+        propertyName = tag.name.getFullText().trim();
+    }
+    return propertyName;
+}
+
+// Create TypeReferenceNode when we reach the leaf node of AST
+function createTypeReference(type: TypeNode): TypeNode | undefined {
+    let typeReference;
+    if (type.kind === SyntaxKind.ParenthesizedType) {
+        type = (type as ParenthesizedTypeNode).type;
+    }
+    // Create TypeReferenceNode for UnionType
+    if (type.kind === SyntaxKind.UnionType) {
+        const elements = (type as UnionTypeNode).types;
+        const nodes = elements.reduce((nodeArray, element) => {
+            const node = transformUnionTypeKeyword(element.kind);
+            if (node) return [...nodeArray, node];
+        }, []);
+        if (!nodes) return;
+        typeReference = factory.createUnionTypeNode(nodes);
+    }
+    //Create TypeReferenceNode for primitive types
+    else {
+        typeReference = transformUnionTypeKeyword(type.kind);
+    }
+    return typeReference;
+}
+
+function transformUnionTypeKeyword(keyword: TypeNodeSyntaxKind): TypeNode | undefined {
+    switch (keyword) {
+        case SyntaxKind.NumberKeyword:
+            return factory.createTypeReferenceNode("number");
+        case SyntaxKind.StringKeyword:
+            return factory.createTypeReferenceNode("string");
+        case SyntaxKind.UndefinedKeyword:
+            return factory.createTypeReferenceNode("undefined");
+        case SyntaxKind.ObjectKeyword:
+            return factory.createTypeReferenceNode("object");
+        case SyntaxKind.VoidKeyword:
+            return factory.createTypeReferenceNode("void");
+        default:
+            return;
+    }
+}
+
+/** @internal */
+export function _containsJSDocTypedef(node: Node): node is HasJSDoc {

```

Comment:
Style nit, but you don't need the underscore prefix here (as far as I'm concerned).

Reply:
fixed

Reply:
Are you marking these fixed as you fix them locally to push later, or did you intend to push? (I'm not seeing pushes, hence me asking.)